app/src/utils/transfer.ts

```diff
@@ -224,3 +224,61 @@ export function toAmountInfo(
     inDollars: tokenAmount.amount * usdPrice,
   }
 }
+
+/**
+ * Orders completed transfers by date.
+ * @param transfers - The list of completed transfers to order.
+ * @returns The list of completed transfers ordered by date.
+ */
+export const orderTransfersByDate = (transfers: CompletedTransfer[]) =>
+  transfers.reduce<TransfersByDate>((acc, transfer) => {
+    let date: string
+    if (typeof transfer.date === 'string') {
+      date = new Date(transfer.date).toISOString().split('T')[0]
+    } else if (transfer.date instanceof Date) {
+      date = transfer.date.toISOString().split('T')[0]
+    } else {
+      date = 'Unknown date'
+    }
+
+    if (!acc[date]) {
+      acc[date] = []
+    }
+    acc[date].push(transfer)
+    return acc
+  }, {})
+
+/**
+ * Formats the ordered completed transfers list to match the transfer history design.
+ * @param transfers - The list of completed transfers to format.
+ * @returns The formatted list of completed transfers.
+ */
+export const formatTransfersByDate = (transfers: CompletedTransfer[]) => {
+  const orderedTransfersByDate = orderTransfersByDate(transfers)
+  return Object.keys(orderedTransfersByDate)
+    .map(date => {
+      return { date, transfers: orderedTransfersByDate[date] }
+    })
+    .sort((a, b) => new Date(b.date).getTime() - new Date(a.date).getTime())
+}
+
+/**
+ * Checks if an ongoing transfer is outdated and should be marked as undefined:
+ * - XCM transfers are considered outdated after 1 hour.
+ * - Bridge transfers are considered outdated after 6 hours.
+ *
+ * @param transfer - The ongoing transfer to check.
+ * @returns A boolean indicating whether the transfer is outdated.
+ */
+export const isTransferStatusUndefined = (
```

Comment:
To play the devil's advocate, this function doesn't check if the tx status is "undefined" or "outdated", it just checks if it was triggered longer than the threshold we convey for each tx type (xcm and bridged) - so this would return true for successful txs for example.

A possible alternative:

```suggestion
export const  startedTooLongAgo = (
```

Reply:
all right